rs_zktrie/src/raw.rs

```diff
@@ -352,68 +352,115 @@ impl<H: Hashable, DB: ZktrieDatabase, const MAX_LEVELS: usize> ZkTrieImpl<H, DB,
             let n = self.get_node(&next_hash)?;
-            //sanity check
-            if i > 0 && n.is_terminal() {
-                if last_node_type == NodeTypeBranch3 {
-                    panic!("parent node has invalid type: children are not terminal")
-                } else if path[i - 1] && last_node_type == NodeTypeBranch1 {
-                    panic!("parent node has invalid type: right child is not terminal")
-                } else if !path[i - 1] && last_node_type == NodeTypeBranch2 {
-                    panic!("parent node has invalid type: left child is not terminal")
+
+            #[cfg(debug_assertions)]
+            {
+                // sanity check
+                if i > 0 && n.is_terminal() {
+                    if last_node_type == NodeTypeBranch3 {
+                        panic!("parent node has invalid type: children are not terminal")
+                    } else if path[i- 1] && last_node_type == NodeTypeBranch1 {
+                        panic!("parent node has invalid type: right child is not terminal")
+                    } else if !path[i - 1] && last_node_type == NodeTypeBranch2 {
+                        panic!("parent node has invalid type: left child is not terminal")
+                    }
                 }
+                last_node_type = n.node_type;
             }
 
-            last_node_type = n.node_type;
             match n.node_type {
                 NodeTypeEmptyNew => {
-                    node = Some(Node::<H>::new_empty_node());
-                    Ok(())
+                    return Ok(Node::<H>::new_empty_node());
                 }
                 NodeTypeLeafNew => {
-                    if *node_key == n.node_key {
-                        node = Some(n);
-                        Ok(())
+                    return if *node_key == n.node_key {
+                        Ok(n)
                     } else {
                         Err(ImplError::ErrKeyNotFound)
                     }
                 }
                 NodeTypeBranch0 | NodeTypeBranch1 | NodeTypeBranch2 | NodeTypeBranch3 => {
                     if path[i] {
                         next_hash = n.child_right.unwrap();
-                        siblings.push(n.child_left.unwrap());
                     } else {
                         next_hash = n.child_left.unwrap();
-                        siblings.push(n.child_right.unwrap());
-                    };
-                    Ok(())
-                }
-                NodeTypeEmpty | NodeTypeLeaf | NodeTypeParent => {
-                    panic!("encounter deprecated node types")
+                    }
                 }
-                _ => Err(ImplError::ErrInvalidNodeFound),
-            }?
+                _ => return Err(ImplError::ErrInvalidNodeFound),
+            }
         }
-        Ok((node.unwrap(), siblings))
+        Err(ImplError::ErrKeyNotFound)
     }
 
+    // fn try_get_with_path(&self, node_key: &H) -> Result<(Node<H>, Vec<H>), ImplError> {
```

Comment:
The optimization is good. Just consider maybe we should keep this old function with another name.

It return both the found node and the path, which is important for mpt proof.

Currently the rs-zktrie do not be used by `l2geth` but i am not sure what would be our future plan like.

Reply:
so I comment out this, since it's a private method and no usage.

Reply:
It is ok. What about just leave some comment so we would not prune it wrongly later?